src/Symfony/Component/Validator/Constraints/FileValidator.php

```diff
@@ -170,12 +172,12 @@ public function validate($value, Constraint $constraint)
         }
 
         if ($constraint->mimeTypes) {
-            if (!$value instanceof FileObject) {
-                $value = new FileObject($value);
+            if (!class_exists(MimeTypes::class)) {
```

Comment:
Won't this break applications where people use the Validator component in 4.4, but still use version 3.4 of the HttpFoundation component (previously that combination was able to successfully validate)?

Reply:
You're right! @pierredup can you please update to account for the case were only HttpFoundation is found?